src/twisted/conch/endpoints.py

```diff
@@ -445,30 +457,29 @@ def verifyHostKey(self, hostKey, fingerprint):
             L{KnownHostsFile.verifyHostKey}.
         """
         hostname = self.creator.hostname
-        ip = networkString(self.transport.getPeer().host)
-
+        transport = self.transport
+        assert transport is not None
+        peer = transport.getPeer()
+        assert isinstance(peer, (IPv4Address, IPv6Address))
+        ip = networkString(peer.host)
```

Comment:
Maybe we can also stop using `networkString` from `compat`

```suggestion
        ip = peer.host.encode("ascii")
```